stepconfig/stepconfig.go

```diff
@@ -0,0 +1,169 @@
+package stepconfig
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"reflect"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/bitrise-io/go-utils/log"
+)
+
+// ErrInvalidConfig indicates that a configuration is of the wrong type.
+var ErrInvalidConfig = errors.New("conf must be a struct pointer")
+
+// ParseError occurs when a struct field cannot be set.
+type ParseError struct {
```

Comment:
I think the ErrInvalidConfig should be a ParseError, since the package can not parse the provided struct:

- nil passed
- non pointer passed

I think this error should be: FieldError, since this error describes an issue with a field